src/linear_algebra/free_algebra.lean

```diff
@@ -12,17 +12,20 @@ import linear_algebra.finsupp_vector_space
 This file provides a `free_monoid X` basis on the `free_algebra R X`, and uses it to show the
 dimension of the algebra is the cardinality of `list X`
 -/
+
+universe variables u v
+
 namespace free_algebra
 
 /-- The `free_monoid X` basis on the `free_algebra R X`,
 mapping `[x₁, x₂, ..., xₙ]` to the "monomial" `1 • x₁ * x₂ * ⋯ * xₙ` -/
 @[simps]
-noncomputable def basis_free_monoid (R : Type*) (X : Type*) [comm_ring R] :
+noncomputable def basis_free_monoid (R : Type u) (X : Type v) [comm_ring R] :
   basis (free_monoid X) R (free_algebra R X) :=
 finsupp.basis_single_one.map
   (equiv_monoid_algebra_free_monoid.symm.to_linear_equiv : _ ≃ₗ[R] free_algebra R X)
 
-lemma dim_eq {K : Type*} {X : Type*} [field K] :
+lemma dim_eq {K : Type u} {X : Type (max u v)} [field K] :
```

Comment:
I'm not sure that `X` should be in a bigger universe here - what goes wrong if it's in `v`?

Reply:
If `X` has type `v` then `free_algebra K X` has level `max u v` and the equality is between cardinals of different universe levels (`max u v` on the left and `v` on the right).

Feel free to make a separate PR (preferably on top of this one) that changes the statement to use `cardinal.lift`

Reply:
Note: I didn't change anything here, just made it explicit:
![image](https://user-images.githubusercontent.com/7376012/129483948-66b71a01-8aeb-4c79-ba54-783b4c03dd27.png)

Reply:
Ah, I guess your point is that the statement always stated what you have here, just implicitly. Perhaps just
```suggestion
/-- TODO: generalize to `X : Type v` -/
lemma dim_eq {K : Type u} {X : Type (max u v)} [field K] :
```